File: prime_number_primer/prime_number_primer__prime_factorization.py
# 実行時間 10ms
i = 2
n = int(input())
l = []

while i * i <= n:
    if n % i != 0:
        i += 1
        continue
    while n % i == 0:
        l.append(i)
        n //= i
    i += 1
if n != 1:
    l.append(n)
print("\n".join(map(str, sorted(l))))

File: prime_number_primer/prime_number_primer__prime_factorization.cs
using System.Linq;
using static System.Console;

internal class Program
{
    static void Main()
    {
        int n = int.Parse( ReadLine() );
        var l = new System.Collections.Generic.List<int>();

        for( int i = 2; i * i <= n; i++ ) {
            if( n % i != 0 ) {
                continue;
            }
            while( n % i == 0 ) {
                l.Add( i );
                n /= i;
            }
        }
        if( n != 1 ) {
            l.Add( n );
        }
        WriteLine( string.Join( System.Environment.NewLine, l.OrderBy( x => x ) ) );
        return;
    }
}
